src/server/table.h

```diff
@@ -99,26 +101,13 @@ class LockTable {
   }
 
  private:
-  struct Key {
-    operator std::string_view() const {
-      return visit([](const auto& s) -> std::string_view { return s; }, val_);
-    }
-
-    bool operator==(const Key& o) const {
-      return *this == std::string_view(o);
-    }
-
-    friend std::ostream& operator<<(std::ostream& o, const Key& key) {
-      return o << std::string_view(key);
+  // We use fingerprinting before accessing locks - no need to mix more.
+  struct Hasher {
+    size_t operator()(uint64_t val) const {
+      return val;
     }
-
-    // If the key is backed by a string_view, replace it with a string with the same value
-    void MakeOwned() const;
-
-    mutable std::variant<std::string_view, std::string> val_;
   };
-
-  absl::flat_hash_map<Key, IntentLock> locks_;
+  absl::flat_hash_map<uint64_t, IntentLock, Hasher> locks_;
```

Comment:
Why do we need a custom hasher instead of just `absl::flat_hash_map<uint64_t, IntentLock>`?

Reply:
line 104 explains. If it's not clear I will expand there.